lib/active_admin/filters/active_sidebar.rb

```diff
@@ -11,32 +11,47 @@ def initialize
 
       def block
         -> do
-          active_filters = ActiveAdmin::Filters::Active.new(active_admin_config, assigns[:search])
-          span do
-            if current_scope
-              h4 I18n.t("active_admin.search_status.current_scope"), style: "display: inline"
-              b scope_name(current_scope), class: "current_scope_name", style: "display: inline"
-            end
-            div style: "margin-top: 10px" do
-              h4 I18n.t("active_admin.search_status.current_filters"), style: "margin-bottom: 10px"
-              ul do
-                if active_filters.filters.blank?
-                  li I18n.t("active_admin.search_status.no_current_filters")
-                else
-                  active_filters.filters.each do |filter|
-                    li filter.html_options do
-                      span do
-                        text_node filter.label
-                      end
-                      b do
-                        text_node to_sentence(filter.values.map { |v| pretty_format(v) })
-                      end
-                    end
-                  end
-                end
+          def scope_block(current_scope)
+            return unless current_scope
+
+            h4 I18n.t("active_admin.search_status.current_scope"), style: "display: inline"
+            b scope_name(current_scope), class: "current_scope_name"
+          end
+
+          def filters_list(active_filters, active_scopes)
+            h4 I18n.t("active_admin.search_status.current_filters")
+            ul do
```

Comment:
@vlad-psh  👍 , what do you think can methods be defined out of the block ?

Reply:
@Fivell Thanks for your comment.

Initially I wanted to make changes as minimal as possible, so I just edited our existing (pretty long) `block` method (just splitted internal lambda with loops into smaller methods for readability; max nesting level were reduced from 13 to just 8)

But I agree, it would be even better to extract those methods outside lambda. So I created new commit adding new view component `active_filters_sidebar_content` to render content for active filters.

Please check, what do you think?

Reply:
@vlad-psh  looks good thanks, lets wait for CI

Reply:
To get the process it would be even better to extract those methods outside lambda. So I created new commit adding new view component active_filters_sidebar_